Services/Simulations.cs

```diff
@@ -281,6 +293,19 @@ public async Task DeleteAsync(string id)
             await this.storage.DeleteAsync(STORAGE_COLLECTION, id);
         }
 
+        public async Task AddDeviceAsync(string id)
```

Comment:
is this needed? the agent can make this call directly

Reply:
this task is internally calling accessing this.storage which is private, so won't be able call it directly.

Reply:
I'm not seeing where storage is used here.  I'm actually suggesting to add this dependency
```
IDevices devices
```
in `Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.Agent` and there, at line 142 replacing
```
await this.simulations.DeleteDevicesAsync(ids);
```
with
```
await this.devices.DeleteListAsync(ids);
```
so this class doesn't need this new method, which is just proxying the call to `IDevices`.

Why wouldn't that work?